src/overlays/gamestates/ovl_file_choose/z_file_choose_NES.c

```diff
@@ -4,96 +4,2362 @@
+    gSaveContext.skyboxTime = 0;
+    gSaveContext.save.time = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/func_80812A6C.s")
+    Skybox_Init(&this->state, &this->skyboxCtx, 1);
+    R_TIME_SPEED = 10;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/func_80812D44.s")
+    envCtx->changeSkyboxState = 0;
+    envCtx->changeSkyboxTimer = 0;
+    envCtx->changeLightEnabled = false;
+    envCtx->changeLightTimer = 0;
+    envCtx->skyboxDmaState = 0;
+    envCtx->skybox1Index = 99;
+    envCtx->skybox2Index = 99;
+    envCtx->lightConfig = 0;
+    envCtx->changeLightNextConfig = 0;
+    envCtx->lightSetting = 0;
+    envCtx->skyboxConfig = 2;
+    envCtx->skyboxDisabled = 0;
+    envCtx->skyboxBlend = 0;
+    envCtx->glareAlpha = 0.0f;
+    envCtx->lensFlareAlphaScale = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/func_80812D94.s")
+    gSaveContext.buttonStatus[EQUIP_SLOT_B] = BTN_ENABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_C_LEFT] = BTN_ENABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_C_DOWN] = BTN_ENABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_C_RIGHT] = BTN_ENABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_A] = BTN_ENABLED;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/func_80812E94.s")
+void FileSelect_Destroy(GameState* this) {
+    ShrinkWindow_Destroy();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/func_80812ED0.s")
+void FileSelect_Init(GameState* thisx) {
+    s32 pad;
+    FileSelectState* this = (FileSelectState*)thisx;
+    size_t size;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/FileSelect_UpdateAndDrawSkybox.s")
+    Game_SetFramerateDivisor(&this->state, 1);
+    Matrix_Init(&this->state);
+    ShrinkWindow_Init();
+    View_Init(&this->view, this->state.gfxCtx);
+    this->state.main = FileSelect_Main;
+    this->state.destroy = FileSelect_Destroy;
+    FileSelect_InitContext(&this->state);
+    Font_LoadOrderedFont(&this->font);
 
-void FileSelect_Main(GameState* thisx);
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/FileSelect_Main.s")
+    size = SEGMENT_ROM_SIZE(title_static);
+    this->staticSegment = THA_AllocTailAlign16(&this->state.heap, size);
+    DmaMgr_SendRequest0(this->staticSegment, SEGMENT_ROM_START(title_static), size);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/func_80813908.s")
+    size = SEGMENT_ROM_SIZE(parameter_static);
+    this->parameterSegment = THA_AllocTailAlign16(&this->state.heap, size);
+    DmaMgr_SendRequest0(this->parameterSegment, SEGMENT_ROM_START(parameter_static), size);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/FileSelect_Destroy.s")
+    size = gObjectTable[OBJECT_MAG].vromEnd - gObjectTable[OBJECT_MAG].vromStart;
+    this->titleSegment = THA_AllocTailAlign16(&this->state.heap, size);
+    DmaMgr_SendRequest0(this->titleSegment, gObjectTable[OBJECT_MAG].vromStart, size);
```

Comment:
Interesting, what does it use from object_mag?

Reply:
Not a clue, a quick search and I don't see anything

Reply:
IIRC some versions don't have this anyway.